### PR TITLE
JAVA-2603: Race condition in change stream tests

### PR DESCRIPTION
**Pull request by jyemin**
Wait for all connections pools to have no checked out connections after closing an asynchronous cursor.

This is basically what's already done in AsyncQueryBatchCursorFunctionalSpecification except that there we know the server address so can just attack that one connection pool, whereas here there's no method to get the server address for the cursor so we have to attack them all.

https://evergreen.mongodb.com/version/5a204f16e3c33173de000849

### CODE REVIEW

File: driver-core/src/test/functional/com/mongodb/operation/AggregateOperationSpecification.groovy [hunk 1]
```diff
@@ -191,7 +193,7 @@ class AggregateOperationSpecification extends OperationFunctionalSpecification {
         async << [true, false]
     }
 
-    @IgnoreIf({ !(serverVersionAtLeast(3, 6) && isDiscoverableReplicaSet()) })
+    @IgnoreIf({ !(serverVersionAtLeast(3, 6) && !isStandalone()) })
```
Comment (jyemin) [inline driver-core/src/test/functional/com/mongodb/operation/AggregateOperationSpecification.groovy]: Should put this in a separate commit.   But the change is because we can test change streams against sharded clusters now and we missed this spot.

Comment (rozza) [reply driver-core/src/test/functional/com/mongodb/operation/AggregateOperationSpecification.groovy]: No its fine here.
